workflows/admin/assembly_analysis_batch.py

```diff
@@ -158,24 +237,109 @@ def mark_for_rerun(self, request, queryset):
         Useful for rerunning failed batches.
         """
         for batch in queryset:
-            batch.asa_status = AssemblyAnalysisPipelineStatus.PENDING
-            batch.virify_status = AssemblyAnalysisPipelineStatus.PENDING
-            batch.map_status = AssemblyAnalysisPipelineStatus.PENDING
-            batch.save()
-
-            # Reset all batch assembly analyses to PENDING
-            # This doesn't account for the disabled ones.
-            batch.batch_analyses.objects.all().update(
+            # Reset all batch-analysis relationships to PENDING (excludes disabled)
+            batch.batch_analyses.update(
                 asa_status=AssemblyAnalysisPipelineStatus.PENDING,
                 virify_status=AssemblyAnalysisPipelineStatus.PENDING,
                 map_status=AssemblyAnalysisPipelineStatus.PENDING,
             )
 
+            # Update counts to reflect a new status
+            batch.update_pipeline_status_counts(AssemblyAnalysisPipeline.ASA)
+            batch.update_pipeline_status_counts(AssemblyAnalysisPipeline.VIRIFY)
+            batch.update_pipeline_status_counts(AssemblyAnalysisPipeline.MAP)
+
         self.message_user(
             request,
             f"Successfully reset {queryset.count()} batch(es) to PENDING status.",
         )
 
+    @action(description="Retry FAILED ASA analyses only")
+    def retry_failed_asa(self, request, queryset):
+        """
+        Reset only FAILED ASA analyses to PENDING status.
+
+        This allows selective retry of failed ASA analyses without reprocessing
+        successful ones. Useful when:
+        - Transient failures occurred (network, disk space)
+        - Input data was fixed for specific analyses
+        - Validation rules were updated
+
+        TODO: the ASA, MAP and VIRify methods are repeated.. I'll refactor this (mbc)
+
+        :param request: Django admin request
+        :param queryset: Selected AssemblyAnalysisBatch objects
+        """
+        total_reset = 0
+        for batch in queryset:
+            count = batch.batch_analyses.filter(
+                asa_status=AssemblyAnalysisPipelineStatus.FAILED
```

Comment:
it would be nice to use the existing helper `SelectByStatusManagerMixin` for the batch models. It has helpers for filtering in or excluding out instances based on a list of matching statuses. Can be done later, just a though.